polaris-core/src/main/java/org/apache/polaris/core/entity/CatalogEntity.java

```diff
@@ -279,6 +285,15 @@ public Builder setStorageConfigurationInfo(
       return this;
     }
 
+    /** Validate the number of allowed locations not exceeding the max value. */
+    public void validateMaxAllowedLocations(
+        Collection<String> allowedLocations, int maxAllowedLocations) {
```

Comment:
Since the config accessor for `maxAllowedLocations` is static, why not call it inside this method as opposed to at call sites?